org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java

```diff
@@ -194,6 +199,8 @@ private static URL makeImageURL(String prefix, String name) {
 	}
 
 	public static Image manage(String key, ImageDescriptor desc) {
+		if (PLUGIN_REGISTRY == null)
```

Comment:
Here I'd rather go for a call to ensureInitialized(); before returning null.
```suggestion
		ensureInitialized();
		if (PLUGIN_REGISTRY == null)
```

Reply:
<img width="455" alt="image" src="https://github.com/eclipse/xtext/assets/1202254/83114350-d27c-430e-b3ff-fa0128bc41eb">

@szarnekow `manage` is called by `initialize`, which is called by `ensureInitialized`, but `ensureInitialize` checks whether PLUGIN_REGISTRY is not null, so there should be no cycle. By the way, `manage` is public but we only use it in `initialize`. I'll apply your suggestion.